src/components/Timer.tsx

```diff
@@ -33,9 +34,67 @@ const H1 = styled.h1<TypographyProps & FlexboxProps>`
 const Span = styled.span<SpaceProps>`
   ${space}
 `;
+
+const launchDates: LaunchDateProps[] = [
```

Comment:
super impressed with this but i would say there is a simpler way to do it.

Whenever you use the `Timer` component pass a `endDate` prop:

```
const Belledejour = () => {
  return (
    <Main justifyContent="center">
      <Timer endDate="2020-11-20"/>
    </Main>
  );
};
```


```
const calculateTimeLeft = () => {
  let year = new Date().getFullYear();
  const difference = +new Date(`${endDate}`) - +new Date();
```

and you can remove all the `const launchDates` and `GetDate` function

Reply:
hmm still wrapping my head around this solution. something in the implementation isn't clicking. gonna walk away and come back to it

Reply:
struggling a bit with this @Jbarget -  I've read loads of TS threads and I kind of understand what the error is telling me, I think it's telling me that the prop can not be read inside of the Timer component because I haven't correctly passed it from parent to child yet.

I'm a bit confused if I'm supposed to be doing `import Timer, {endDate} from "../Timer";` just like we do with Styled Component props, but this isn't compiling for me. 

Can you tell me where I'm going wrong?

Reply:
https://www.robinwieruch.de/react-pass-props-to-component

Reply:
https://medium.com/innovation-and-technology/deciphering-typescripts-react-errors-8704cc9ef402

Reply:
On the `Timer` component you need to declare which props it receives.

Eg. something you've used `const Component: React.FC<ComponentProps> = () => {}`